src/daft-sql/src/table_provider/read_json.rs

```diff
@@ -0,0 +1,68 @@
+use daft_scan::builder::JsonScanBuilder;
+
+use super::{expr_to_iocfg, SQLTableFunction};
+use crate::{error::PlannerError, functions::SQLFunctionArguments};
+
+pub(super) struct ReadJsonFunction;
+
+impl SQLTableFunction for ReadJsonFunction {
+    fn plan(
+        &self,
+        planner: &crate::SQLPlanner,
+        args: &sqlparser::ast::TableFunctionArgs,
+    ) -> crate::error::SQLPlannerResult<daft_logical_plan::LogicalPlanBuilder> {
+        let builder: JsonScanBuilder = planner.plan_function_args(
+            args.args.as_slice(),
+            &[
+                "path",
+                "infer_schema",
+                // "schema"
+                "io_config",
+                "file_path_column",
+                "hive_partitioning",
+                // "schema_hints",
+                "buffer_size",
+                "chunk_size",
+            ],
+            1, // (path)
+        )?;
+        let runtime = common_runtime::get_io_runtime(true);
+        let result = runtime.block_on(builder.finish())??;
+        Ok(result)
+    }
+}
+
+impl TryFrom<SQLFunctionArguments> for JsonScanBuilder {
+    type Error = PlannerError;
+
+    fn try_from(args: SQLFunctionArguments) -> Result<Self, Self::Error> {
+        // TODO validations (unsure if should carry over from python API)
+        // - schema_hints is deprecated
+        // - ensure infer_schema is true if schema is None.
+
+        let glob_paths: String = args
+            .try_get_positional(0)?
+            .ok_or_else(|| PlannerError::invalid_operation("path is required for `read_json`"))?;
```

Comment:
not necessary in this PR, but I think it'd be nice to also support an array of paths here _(we support this in the dataframe API)_

something like 
```sql
select * from read_json(['./file1.json', './file2.json'])
```

Reply:
Here's an issue for all three to support the array of paths.

https://github.com/Eventual-Inc/Daft/issues/3686